colony/commands/sb.py

```diff
@@ -151,63 +190,70 @@ def do_start(self):
+        branch_name_or_type = ""
+        if working_branch:
+            branch_name_or_type = working_branch + "-"
+        if temp_working_branch:
+            branch_name_or_type = "localchanges-"
+        return f"{blueprint_name}-{branch_name_or_type}{suffix}"
+
+
+def wait_for_sandbox_to_launch(
+    sb_manager: SandboxesManager,
+    sandbox_id: str,
+    timeout: int,
+    context_branch: ContextBranch,
+    wait: bool,
+) -> bool:
+
+    if not wait and not context_branch.temp_branch_exists:
+        return False
+    try:
+        if context_branch.temp_branch_exists:
+            context_branch.revert_from_local_temp_branch()
+
+        if not timeout:
+            timeout = DEFAULT_TIMEOUT
+
+        start_time = datetime.datetime.now()
+        sandbox = sb_manager.get(sandbox_id)
+        status = getattr(sandbox, "sandbox_status")
 
-        branch_to_be_used = temp_working_branch or working_branch
-
-        if name is None:
-            suffix = datetime.datetime.now().strftime("%b%d-%H:%M:%S")
-            branch_name_or_type = ""
-            if working_branch:
-                branch_name_or_type = working_branch + "-"
-            if temp_working_branch:
-                branch_name_or_type = "localchanges-"
-            name = f"{blueprint_name}-{branch_name_or_type}{suffix}"
-
-        try:
-            sandbox_id = self.manager.start(
-                name, blueprint_name, duration, branch_to_be_used, commit, artifacts, inputs
-            )
-            BaseCommand.action_announcement("Starting sandbox")
-            BaseCommand.important_value("Id: ", sandbox_id)
-            BaseCommand.url(prefix_message="URL: ", message=self.manager.get_sandbox_ui_link(sandbox_id))
-
-        except Exception as e:
-            logger.exception(e, exc_info=False)
-            if temp_working_branch.startswith(UNCOMMITTED_BRANCH_NAME):
-                revert_from_temp_branch(repo, working_branch, stashed_flag)
-                delete_temp_branch(repo, temp_working_branch)
-            return self.die()
-
-        # todo: I think the below can be simplified and refactored
-        if timeout is None:
-            revert_wait_and_delete_temp_branch(
-                self.manager, blueprint_name, repo, sandbox_id, stashed_flag, temp_working_branch, working_branch
-            )
-            return self.success("The Sandbox was created")
-
-        else:
-            start_time = datetime.datetime.now()
+        sandbox_start_wait_output(sandbox_id, context_branch.temp_branch_exists)
 
-            logger.debug(f"Waiting for the Sandbox {sandbox_id} to start...")
-            # Waiting loop
+        with yaspin(text="Starting...", color="yellow") as spinner:
             while (datetime.datetime.now() - start_time).seconds < timeout * 60:
-                sandbox = self.manager.get(sandbox_id)
+                if status in FINAL_SB_STATUSES:
+                    spinner.green.ok("✔")
```

Comment:
maybe we can show a red "X" when we get a bad final status?

Reply:
It can be easily done... by problem with it is that it wouldn`t be consistent in case we exit waiter ahead of final state...
so if temp_branch can be deleted and no wait flag set....we would just print green V ok